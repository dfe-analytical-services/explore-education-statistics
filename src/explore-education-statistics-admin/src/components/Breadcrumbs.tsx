import { dashboardRoute } from '@admin/routes/routes';
import React from 'react';
import Link from './Link';

export interface BreadcrumbsProps {
  breadcrumbs?: {
    link?: string;
    name: string;
  }[];
  homePath?: string;
}

const Breadcrumbs = ({
  breadcrumbs = [],
  homePath = dashboardRoute.path,
}: BreadcrumbsProps) => {
  const currentBreadcrumbIndex = breadcrumbs.length - 1;

  return (
    <div className="govuk-breadcrumbs">
      <ol className="govuk-breadcrumbs__list" data-testid="breadcrumbs--list">
        {homePath && (
          <li className="govuk-breadcrumbs__list-item">
            <Link className="govuk-breadcrumbs__link" to={homePath}>
              Home
            </Link>
          </li>
        )}
        {breadcrumbs.map((breadcrumb, index) => {
          return (
            <li
              key={`${breadcrumb.name}_${breadcrumb.link}`}
              className="govuk-breadcrumbs__list-item"
            >
              {index < currentBreadcrumbIndex && breadcrumb.link ? (
                <Link className="govuk-breadcrumbs__link" to={breadcrumb.link}>
                  {breadcrumb.name}
                </Link>
              ) : (
                breadcrumb.name
              )}
            </li>
          );
        })}
      </ol>
    </div>
  );
};

export default Breadcrumbs;
