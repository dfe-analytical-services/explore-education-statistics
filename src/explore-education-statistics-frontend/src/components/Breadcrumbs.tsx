import React from 'react';
import Link from './Link';

export interface BreadcrumbsProps {
  breadcrumbs?: {
    link?: string;
    name: string;
  }[];
}

const Breadcrumbs = ({ breadcrumbs = [] }: BreadcrumbsProps) => {
  const currentBreadcrumbIndex = breadcrumbs.length - 1;
  return (
    <nav className="govuk-breadcrumbs" aria-label="Breadcrumb">
      <ol className="govuk-breadcrumbs__list" data-testid="breadcrumbs--list">
        <li className="govuk-breadcrumbs__list-item">
          <Link className="govuk-breadcrumbs__link" to="/">
            Home
          </Link>
        </li>
        {breadcrumbs.map((breadcrumb, index) => {
          return (
            <li
              key={`${breadcrumb.name}_${breadcrumb.link}`}
              className="govuk-breadcrumbs__list-item"
            >
              {index < currentBreadcrumbIndex && breadcrumb.link ? (
                <Link className="govuk-breadcrumbs__link" to={breadcrumb.link}>
                  {breadcrumb.name}
                </Link>
              ) : (
                breadcrumb.name
              )}
            </li>
          );
        })}
      </ol>
    </nav>
  );
};

export default Breadcrumbs;
