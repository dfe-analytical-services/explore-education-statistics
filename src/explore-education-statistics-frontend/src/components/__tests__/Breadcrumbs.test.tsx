import React from 'react';
import { render } from '@testing-library/react';
import Breadcrumbs from '@frontend/components/Breadcrumbs';

describe('Breadcrumbs', () => {
  test('renders correctly with just home breadcrumb', () => {
    const { container } = render(<Breadcrumbs breadcrumbs={[]} />);

    const breadcrumbs = container.querySelectorAll('li');

    expect(breadcrumbs).toHaveLength(1);
    expect(breadcrumbs[0].textContent).toBe('Home');
    expect(breadcrumbs[0].querySelector('a')).toHaveAttribute('href', '/');

    expect(container.innerHTML).toMatchSnapshot();
  });

  test('renders correctly with multiple breadcrumbs', () => {
    const { container } = render(
      <Breadcrumbs
        breadcrumbs={[
          {
            link: '/publications',
            name: 'Publications',
          },
          {
            link: '/test-publication',
            name: 'Test publication',
          },
        ]}
      />,
    );

    const breadcrumbs = container.querySelectorAll('li');

    expect(breadcrumbs).toHaveLength(3);
    expect(breadcrumbs[0].textContent).toBe('Home');
    expect(breadcrumbs[0].querySelector('a')).toHaveAttribute('href', '/');
    expect(breadcrumbs[1].textContent).toBe('Publications');
    expect(breadcrumbs[1].querySelector('a')).toHaveAttribute(
      'href',
      '/publications',
    );
    expect(breadcrumbs[2].textContent).toBe('Test publication');

    expect(container.innerHTML).toMatchSnapshot();
  });

  test('does not render a link if breadcrumb is missing a link', () => {
    const { container } = render(
      <Breadcrumbs
        breadcrumbs={[
          {
            name: 'Publications',
          },
          {
            link: '/test-publication',
            name: 'Test publication',
          },
        ]}
      />,
    );

    const breadcrumbs = container.querySelectorAll('li');

    expect(breadcrumbs).toHaveLength(3);
    expect(breadcrumbs[1].textContent).toBe('Publications');
    expect(breadcrumbs[1].querySelector('a')).toBe(null);
  });

  test('does not render last breadcrumb as a link', () => {
    const { container } = render(
      <Breadcrumbs
        breadcrumbs={[
          {
            link: '/publications',
            name: 'Publications',
          },
          {
            link: '/test-publication',
            name: 'Test publication',
          },
        ]}
      />,
    );

    const lastBreadcrumb = container.querySelector(
      'li:last-child',
    ) as HTMLElement;

    expect(lastBreadcrumb).toBeDefined();
    expect(lastBreadcrumb.querySelector('a')).toBeNull();
  });
});
