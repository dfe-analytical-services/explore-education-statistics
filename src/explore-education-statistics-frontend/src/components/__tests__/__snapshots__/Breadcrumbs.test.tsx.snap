// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Breadcrumbs renders correctly with just home breadcrumb 1`] = `
<nav class="govuk-breadcrumbs"
     aria-label="Breadcrumb"
>
  <ol class="govuk-breadcrumbs__list"
      data-testid="breadcrumbs--list"
  >
    <li class="govuk-breadcrumbs__list-item">
      <a href="/"
         class="govuk-link govuk-breadcrumbs__link"
      >
        Home
      </a>
    </li>
  </ol>
</nav>
`;

exports[`Breadcrumbs renders correctly with multiple breadcrumbs 1`] = `
<nav class="govuk-breadcrumbs"
     aria-label="Breadcrumb"
>
  <ol class="govuk-breadcrumbs__list"
      data-testid="breadcrumbs--list"
  >
    <li class="govuk-breadcrumbs__list-item">
      <a href="/"
         class="govuk-link govuk-breadcrumbs__link"
      >
        Home
      </a>
    </li>
    <li class="govuk-breadcrumbs__list-item">
      <a href="/publications"
         class="govuk-link govuk-breadcrumbs__link"
      >
        Publications
      </a>
    </li>
    <li class="govuk-breadcrumbs__list-item">
      Test publication
    </li>
  </ol>
</nav>
`;
